pyproject.toml

```
@@ -26,7 +26,7 @@ grpcio = "^1.48.1"
 grpcio-tools = "^1.48.1"
 google-cloud = "^0.34.0"
 google-api-core = "^2.11.0"
-protobuf = ">=3.20,<5.0"
+protobuf = "^5.26.1"
```

```suggestion
protobuf = ">=3.20,<6.0"
```

let's not hardcode to `5.26.1`, rather put some bound values? will the above work?

---

@vigith this does not mean it is hard-coded. This means that it will take any version `>=5.26.1` and `<6.0`
Nevertheless, I would recommend having a stricter constraint.

```suggestion
protobuf = "~5.26.1"
```
which translates to `>=5.26.1` and `<5.27`

---

let's put `>=3.20,<5.27` then?

---

Sure, but assuming that things would still work with previous versions of protobuf (<5.0) that we support.

---

since protobuf follows semantic versioning, I am assuming that they won't break anything till they reach `6.0`. Now the question is, is 3.x and 4.x still compatible with 5.x, if not we have to bound ourselves to strictly 5.x.

Since our tests were working with earlier version, i am gonna assume that we are good with 3.x and 4.x.

@kohlisid could you please verify these "assumptions"? 🙂

---

Also, I think it's better to set it as `>=3.20,<6.0`, breaking changes are not released in minor upgrades and knowing that protobuf is a library used by millions - it's even more unlikely that any breaking change will be released with minor upgrade.

---

IIRC during our early version of pynumaflow, we used the 3.2 version.

---

Hey @rishhavv 
Were you able to verify the compatibility?